src/components/parts/nav/Search.astro

```diff
@@ -0,0 +1,182 @@
+---
+---
+
+<div id="dummy-search-field" class="field pt-2 pr-4">
+    <p class="control has-icons-right">
+        <input aria-label="search" class="input" type="text" placeholder="Search" style="border: 0" />
+        <span class="icon is-small is-right is-clickable">
+            <i class="fas fa-magnifying-glass"></i>
+        </span>
+    </p>
+</div>
+
+<div id="search-modal" class="modal">
+    <div id="search-box" class="box" style="z-index: 1; width: 60vw">
+        <div class="field pt-2 pr-4">
+            <p class="control has-icons-right is-expanded">
+                <input id="search-field" autocomplete="off" aria-label="search" class="input is-medium" type="text" placeholder="Search" />
+                <span class="icon is-small is-right is-clickable">
+                    <i class="fas fa-magnifying-glass"></i>
+                </span>
+             </p>
+        </div>
+
+        <p id="search-loading" class="has-text-centered is-hidden">Loading...</p>
+        <p id="search-hint" class="has-text-centered">Start typing to search</p>
+        <p id="search-no-result" class="has-text-centered is-hidden">No results found</p>
+
+        <div id="search-result-container">
+        </div>
+    </div>
+    <div id="search-modal-background" class="modal-background"></div>
+</div>
+
+<!-- Dummy element to make sure the two classes don't get optimized away -->
+<div class="has-background-info has-background-link is-hidden"></div>
+
+<script>
+    import lunr from "lunr";
+
+    let isPopupOpen = false;
+    let isIndexReady = false;
+    let isIndexLoading = false;
+    let index = null;
+    let rawData = null;
+
+    const SNIPPET_LENGTH = 256;
+
+    function gid(id) {
+        return document.getElementById(id);
+    }
+
+    async function loadIndex() {
+        if (isIndexLoading) {
+            return;
+        }
+        isIndexLoading = true;
+
+        gid('search-loading').classList.remove('is-hidden');
+
+        await fetch('/search-index.json')
+            .then(response => response.json())
+            .then(data => {
+                rawData = data;
+
+                index = lunr(function () {
+                    this.ref("url");
+                    this.field("title", { boost: 10 });
+                    this.field("content");
+                    this.metadataWhitelist = ['position']
+
+                    for (const url in data) {
+                        this.add({
+                            url: url,
+                            title: data[url].title,
+                            content: data[url].content
+                        });
+                    }
+                });
+
+                gid('search-loading').classList.add('is-hidden');
+                isIndexReady = true;
+            });
+    }
+
+    function highlight(text, startIndex, match) {
+        const highlightColor = (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) ? 'link' : 'warning';
```

Comment:
Isn't this the kind of thing CSS variables are made for ?

Reply:
there isn't a better way to go about it without adding additional CSS to our bulma style

Reply:
sure, I just can't be bothered honestly, I would have to yell at way too many people to get this done considering it needs to be both merged and published

Reply:
Wait it has to go in the quilt-bulma repository ? We can't have custom styles here ?

Reply:
Nope, the CSS is built there, we don't have access to variable here. That's why I hate it.

Reply:
given that the rest of the site has similar issues, I'd say it is fine for now and can be addressed in a future PR
(on that note I *really* would like a theme chooser instead of relying on the browser's preferred color scheme ONLY..)

Reply:
pain²